### DISCUSSION TITLE
writing dwg not working, System.NullReferenceException

### DISCUSSION THREAD

**Discussion by dhulsegge** [DomCR/ACadSharp#122] [Q&A] [answered]
Hi, I'm trying to read a DWG, edit Mtext objects and export to DWG again. 
I've edited your example, but things don't seem to work, do you have any idea what it can be?
https://github.com/DomCR/ACadSharp/blob/master/ACadSharp.Examples/IOExamples.cs
![image](https://user-images.githubusercontent.com/50334651/231433058-30714661-1d95-4173-997e-67cd8fccff15.png)

**Comment by DJGosnell** [comment]
If you could post the stack trace that would be helpful here.

**Reply by dhulsegge** [reply]
Is this useful ?
![image](https://user-images.githubusercontent.com/50334651/231463857-4c31b82c-86c6-4df1-8115-0712323df6fb.png)
i guess it has something to do with the dictionary<ulong,long> map

**Reply by dhulsegge** [reply, marked as answer]
i downloaded the whole repo now manually, instead of the nuget package, i could not write in my c:/ directory... 
I can get copy a DWG now but many objects are lost during copying. Would you like my DWG for further testing? Its an export from Revit as DWG 2007 version.

**Reply by DomCR** [reply]
I would appreciate a copy of the file, it would help with the development.
Can you list some of the objects that are missing in the new file to open new branches for a future implementaion?

Thanks!!

**Reply by dhulsegge** [reply]
a added this code to make a export otherwise it was collapsing already before.
<img width="485" alt="image" src="https://user-images.githubusercontent.com/50334651/231517009-f8e1a593-76c4-47be-8b94-4bb73bb5ce27.png">

**Reply by dhulsegge** [reply]
i think revit creates a lot of blocks during export.
this is the original:
<img width="696" alt="image" src="https://user-images.githubusercontent.com/50334651/231517525-b574a600-4ad5-44a2-bb30-f7744a2bec93.png">
<img width="280" alt="image" src="https://user-images.githubusercontent.com/50334651/231517946-582355e5-be90-45ac-a863-544feea473de.png">

and this is after copying: 
<img width="972" alt="image" src="https://user-images.githubusercontent.com/50334651/231518041-abbf05b5-4495-4d2c-ab61-eab68615a9c1.png">

**Reply by dhulsegge** [reply]
to which email address can i send the file?

**Reply by DomCR** [reply]
In your code you are adding an event in the writer to notify messages into the console, you should be able to get the missing entities, one of them should be `Insert`.

By looking at the changes in your code I know why is failing, the file is not needed thanks I can work with this.

The branch to solve the insert clone problem is this one: https://github.com/DomCR/ACadSharp/tree/implement-IClonable
I'll open another one to implement the Insert writer for dwg, in the mean time you can try to save the file in a dxf.

**Reply by DomCR** [reply]
In your code you are adding an event in the writer to notify messages into the console, you should be able to get the missing entities, one of them should be `Insert`.

By looking at the changes in your code I know why is failing, the file is not needed thanks I can work with this.

The branch to solve the insert clone problem is this one: https://github.com/DomCR/ACadSharp/tree/implement-IClonable
I'll open another one to implement the Insert writer for dwg, in the mean time you can try to save the file in a dxf.

**Reply by dhulsegge** [reply]
DXF gives the same result 
<img width="763" alt="image" src="https://user-images.githubusercontent.com/50334651/231535380-de3084a1-7a47-49c3-8784-896b88c7d274.png">

**Reply by DomCR** [reply]
Thanks for the test, I'll look into that
